hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -536,6 +550,27 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		return ctrl.Result{}, nil
 	}
 
+	// Reconcile PowerVSImage only for the PowerVS platform
+	if nodePool.Spec.Platform.Type == hyperv1.PowerVSPlatform {
+		powervsImage, region, err := getPowerVSImage(hcluster.Spec.Platform.PowerVS.Region, releaseImage)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// Reconcile (Platform)MachineTemplate.
+		image, mutateImage, _, err := ibmPowerVSImageBuilder(hcluster, nodePool, infraID, region, powervsImage)
```

Comment:
this abstraction makes sense for InfraMachineTemplates as we don't know the type ahead of type but it has no value here since you are discriminating by `if nodePool.Spec.Platform.Type == hyperv1.PowerVSPlatform {` all you need is a simple `reconcileIBMPowerVSImage`

Also can you add a comment explaining why this is needed? what is IBMPowerVSImage? and what are the expectations?